src/root_object.jl

```diff
@@ -50,3 +50,42 @@ show(io::IO, rt::Root) = print(io, "Root($(rt.interval), :$(rt.status))")
 ⊆(a::Root, b::Root) = a.interval ⊆ b.interval
 
 big(a::Root) = Root(big(a.interval), a.status)
+
+"""
+    Base.iterate(r::Root{T})
+
+    Implements the first necessary method for iterate over a Root object.
+
+    Outputs:
+    - `r.interval`: interval where some true root of a given function lies.
+    - `state`: indicates that the first iterable element of r has been returned.
+
+"""
+function Base.iterate(r::Root{T}) where {T}
+    state = 1
+    return (r.interval, state)
+end
+
+"""
+    Base.iterate(r::Root{T}, state)
+
+    Implements the second necessary method for iterate over a Root object.
+
+    Inputs:
+    - `r`: Root object.
+    - `state`: indicates the prior element iterated.
+
+    Outputs:
+    - `r.status`: status of the root.
+    - `state`: indicates that the second iterable element of r has been returned.
+    or 
+    - nothing: the iteration has been completed.
+"""
+function Base.iterate(r::Root{T}, state::Int64) where {T}
```

Comment:
Thanks @edtrelo for your contribution.

Following on [this comment](https://github.com/JuliaIntervals/IntervalRootFinding.jl/pull/190#issuecomment-1591086103), I think the error is due to this line: it imposes `state` to be `Int64`. I think the problem is solved if `state` is required to be `Integer`

Reply:
Thanks for the sugguestion @lbenet. I've alredy changed the type of `state`.